src/sentry/utils/concurrent.py

```diff
@@ -92,10 +92,9 @@ def cancel(self, *args, **kwargs):
 
     def set_result(self, *args, **kwargs):
         with self._condition:
-            # This method always overwrites the result, so we always overwrite
-            # the timing, even if another timing was already recorded.
+            result = super().set_result(*args, **kwargs)
             self.__timing[1] = time()
```

Comment:
Should we record `time()` in a local variable before we call `set_result`? That will replicate the existing behaviour.

Reply:
Ah yeah, that's a good suggestion.